src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,296 @@
+
+            String[] command = inputLine.split(" ", COMMAND_TASK_SEPARATOR);
+
+            switch (command[0].toLowerCase()) {
+            case "list":
+                printListMessage(tasks);
+                break;
+            case "done":
+                try {
+                    int doneTaskNumber = Integer.parseInt(command[1]);
+                    task = tasks.get(doneTaskNumber - 1);
+                    task.markAsDone();
+                    printDoneMessage(task);
+                } catch (IndexOutOfBoundsException e) {
+                    printNonExistentTaskMessage();
+                } catch (NumberFormatException e) {
+                    printNotANumberMessage();
+                }
+                break;
+            case "deadline":
+                try {
+                    description = command[1].split("/by", COMMAND_TASK_SEPARATOR);
+                    checkForValidDeadlineInput(description);
+                    Deadline deadline = new Deadline(description[0].trim(), description[1].trim());
+                    tasks.add(deadline);
+                    printAddedMessage(tasks, deadline);
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    printMissingFieldsMessage();
+                } catch (DescriptionFieldEmptyException e) {
+                    printDescriptionFieldEmptyMessage();
+                } catch (TimeFieldEmptyException e) {
+                    printTimeFieldEmptyMessage();
+                } catch (MultipleTimeFieldsException e) {
+                    printTooManyTimeFieldsMessage();
+                }
+                break;
+            case "event":
+                try {
+                    description = command[1].split("/at", COMMAND_TASK_SEPARATOR);
+                    checkForValidEventInput(description);
+                    Event event = new Event(description[0].trim(), description[1].trim());
+                    tasks.add(event);
+                    printAddedMessage(tasks, event);
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    printMissingFieldsMessage();
+                } catch (DescriptionFieldEmptyException e) {
+                    printDescriptionFieldEmptyMessage();
+                } catch (TimeFieldEmptyException e) {
+                    printTimeFieldEmptyMessage();
+                } catch (MultipleTimeFieldsException e) {
+                    printTooManyTimeFieldsMessage();
+                }
+                break;
+            case "todo":
+                try {
+                    ToDo toDo = new ToDo(command[1]);
+                    tasks.add(toDo);
+                    printAddedMessage(tasks, toDo);
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    printMissingFieldsMessage();
+                }
+                break;
+            case "delete":
+                try {
+                    int deleteTaskNumber = Integer.parseInt(command[1]);
```

Comment:
Consider renaming `deleteTaskNumber` to a noun (like `taskNumberToBeDeleted`) rather than an action word.